src/Pauser.sol

```diff
@@ -0,0 +1,332 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.21;
+
+import "@openzeppelin-upgradeable/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin-upgradeable/contracts/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin-upgradeable/contracts/access/OwnableUpgradeable.sol";
+
+import "./interfaces/IPausable.sol";
+import "./RoleRegistry.sol";
+
+contract Pauser is Initializable, UUPSUpgradeable, OwnableUpgradeable {
+    // ========================================= STATE =========================================
+
+    /**
+     * @notice List of contracts that can be paused and unpaused using:
+     * - `pauseAll`
+     * - `unpauseAll`
+     */
+    IPausable[] public pausables;
+
+    /**
+     * @notice Used to pause calls to `deposit` and `depositWithPermit`.
+     */
+    bool public isPaused;
+
+    /**
+     * @notice Minimum delay required to execute an unpause action.
+     */
+    uint256 public minUnpauseDelay;
```

Comment:
in which scenario do we need a delay to unpause a contract?

Reply:
I thought that it was a functional requirement of this task

Reply:
Darn I think we had a miscommunication bc I thought this was a functional requirement of this task :(
 
<img width="1001" alt="Screenshot 2024-07-19 at 10 50 00 AM" src="https://github.com/user-attachments/assets/0f2f5a58-dcf1-4d7b-b1ec-0f305ee78800">

Reply:
oh, don't feel so. I know you were working on it and thought we need that (why not, cause it is more generic). but I was imagining some use cases :)